storage/innobase/btr/btr0cur.cc

```diff
@@ -2182,14 +2182,17 @@ btr_cur_ins_lock_and_undo(
 			GAP-locking we mark that this transaction
 			is using unique key scan here. */
 			if ((type & (DICT_CLUSTERED | DICT_UNIQUE)) == DICT_UNIQUE
-			    && trx->is_wsrep()
-			    && wsrep_thd_is_BF(trx->mysql_thd, false)) {
-				trx->wsrep = 3;
+			    && trx->is_wsrep_BF()) {
+				trx->wsrep_begin_UK_scan();
 			}
 #endif /* WITH_WSREP */
-			if (dberr_t err = lock_rec_insert_check_and_lock(
-				    rec, btr_cur_get_block(cursor),
-				    index, thr, mtr, inherit)) {
+			dberr_t err = lock_rec_insert_check_and_lock(
+					rec, btr_cur_get_block(cursor),
+					index, thr, mtr, inherit);
+#ifdef WITH_WSREP
+			trx->wsrep_end_UK_scan();
+#endif /* WITH_WSREP */
```

Comment:
The indentation of the continuation lines is one TAB too deep.

Could we define `trx_t::wsrep_end_UK_scan()` as dummy empty functions `#ifndef WITH_WSREP`? That would remove some `#ifdef` pollution.